apps/dashboard/src/components/subscribers/hooks/use-navigate-create-subscriber-page.ts

```diff
@@ -0,0 +1,17 @@
+import { useCallback } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+import { useEnvironment } from '@/context/environment/hooks';
+import { buildRoute, ROUTES } from '@/utils/routes';
+
+export const useNavigateCreateSubscriberPage = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { currentEnvironment } = useEnvironment();
+
+  return useCallback(() => {
+    navigate(
+      `${buildRoute(ROUTES.CREATE_SUBSCRIBER, { environmentSlug: currentEnvironment?.slug || '' })}${location.search}`
+    );
+  }, [location.search, navigate, currentEnvironment?.slug]);
```

Comment:
the hook navigates to the subscriber create page and preserves all query params for pagination